lib/repository.js

```diff
@@ -729,15 +731,34 @@ Repository.prototype.discardLines =
   var fullFilePath = path.join(repo.workdir(), filePath);
   var index;
   var originalContent;
+  var filterList;
 
   return repo.refreshIndex()
     .then(function(indexResult) {
       index = indexResult;
+      return FilterList.load(
+        repo,
+        null,
+        filePath,
+        Filter.MODE.CLEAN,
+        Filter.FLAG.DEFAULT
+      );
+    })
+    .then(function(_filterList) {
+      filterList = _filterList;
+
+      if (filterList) {
+        return filterList.applyToFile(repo, filePath);
+      }
 
       return fse.readFile(fullFilePath, "utf8");
     })
     .then(function(content) {
       originalContent = content;
+      if (filterList) {
+        filterList.free();
```

Comment:
My first inclination was to express how we shouldn't do this, and that we should strive to improve nodegit's automatic memory management... My second inclination was that we should really throw away the automatic memory management in nodegit and just expose the free methods on anything that needs freeing and leave memory management up to the consumer of nodegit. It's a little more cognitive on the programmer, but it is _way_ more straightforward and potentially complete, compared to what we have now.

Reply:
You think I should change it now just to be consistent?

Reply:
No. I am pretty certain we're going to move away from our current garbage collection system.